packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -660,21 +661,22 @@ export function gid(id: string | undefined | null): string | null | undefined {
     if (isNull(id)) {
         return null;
     }
-    if (isLightRenderModeVM(vmBeingRendered!)) {
-        return StringReplace.call(id, /\S+/g, (id) => `${id}-${vmBeingRendered.idx}`);
+    const { idx, renderMode } = vmBeingRendered;
+    if (renderMode === RenderMode.Shadow) {
```

Comment:
Shouldn't this run only for synthetic shadow DOM?

Reply:
That's the next PR :)